csharp/src/Google.Protobuf/Google.Protobuf.csproj

```diff
@@ -22,15 +22,30 @@
     <AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'net45' or '$(TargetFramework)' == 'netstandard2.0' ">
-    <DefineConstants>$(DefineConstants);GOOGLE_PROTOBUF_SUPPORT_SYSTEM_MEMORY</DefineConstants>
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net45' ">
+    <DefineConstants>$(DefineConstants);GOOGLE_PROTOBUF_SUPPORT_SYSTEM_MEMORY;PROTOBUF_DISABLE_BUFFER_SERIALIZATION</DefineConstants>
```

Comment:
I think we could set LangVersion to 7.2 and enable GOOGLE_PROTOBUF_SUPPORT_SPAN, at which point we could support buffer serialization in net45 as well?
If we did that, we wouldn't potentially need two separate defines `GOOGLE_PROTOBUF_SUPPORT_SYSTEM_MEMORY`, `GOOGLE_PROTOBUF_SUPPORT_SPAN` that TBH are a bit confusing to me  - it looks like we're using `GOOGLE_PROTOBUF_SUPPORT_SPAN` basically as a proxy for  "supports ref structs"?